swarm.go

```diff
@@ -201,20 +202,11 @@ func (s *Swarm) Conns() []*Conn {
 }
 
 // ConnsWithGroup returns all the connections with a given Group
-//
-// TODO: the primary use of this method is to check if there is a connection
-// to a given peer. That is extremely wasteful in terms of cpu and
-// allocations. This method is optimized for that specific usecase without
-// restructuring much of how go-peerstream works. Realistically, we need to
-// refactor the s.conns field to be a map from peer.ID to *Conn, but this
-// should do for now.
 func (s *Swarm) ConnsWithGroup(g Group) []*Conn {
 	s.connLock.RLock()
 	conns := make([]*Conn, 0, 1)
-	for c := range s.conns {
-		if c.InGroup(g) {
-			conns = append(conns, c)
-		}
+	for c := range s.connIdx[g] {
+		conns = append(conns, c)
```

Comment:
I would change it to:
```go
cs := c.connIdx[g]
conns := make([]*Conn, len(cs))
copy(conns, cs)
```

Reply:
Fixed (it's a map so I still have to iterate but good catch on the allocation size).

Reply:
Ahh, I thought it was an array.